accepted/2021/preview-features/preview-features.md

```diff
@@ -0,0 +1,418 @@
+this issue. Adriana does a quick web search and finds a blog post from the C#
+team that showcases how this feature might look like. She wonders how she can
+play with that. Optimistic, she copy & pastes the sample code into her project
+which immediately generates a compilation error:
+
+> The feature 'static interface members' is currently in Preview and
+> "unsupported". To use Preview features, use the 'preview' language version.
+
+The error displays a light-bulb in Visual Studio, suggesting to enable preview
+features, which modifies her project file as follows:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <EnablePreviewFeatures>True</EnablePreviewFeatures>
+  </PropertyGroup>
+
+</Project>
+```
+
+### Enabling a cross-cutting preview feature
+
+Ainsley is working for the insurance company Fabrikam. As part of their work,
+they are trying to figure out how they can integrate machine learning with
+ML.NET more tightly with their services, but they struggle with data ingestion
+because a lot of the heavy computations are happening inside of an old C++ base.
+When that code was written, .NET wasn't fast enough to do heavy numerical
+computations.
+
+As part of their research into this problem space, they notice a recent blog
+post about how the upcoming generic numeric support and hardware intrinsics can
+help to speed things up considerably. They decide to give it a spin, so they
+copy & paste some sample code from the blog post.
+
+The code immediately produces an error message:
+
+> error: The type 'IArithmetic\<T>' is in preview. In order to use it, you need
+> to enable preview features.
+
+After taking a quick look at the blog post, Ainsley realizes that they only
+needs to set `EnablePreviewFeatures` to `true` in the project file.
+
+### Consuming a library that uses a preview feature
+
+Sahibi works with Ainsley and tries to consume their library that exposes some
+of the core algorithms so she can integrate those with ML.NET. After adding the
+reference to Ainsley's library `Fabrikam.Algorithms.Core` she instantiates the
+engine type which produces the following error:
+
+> The library 'Fabrikam.Algorithms.Core' uses preview features of the .NET
+> platform. In order to use it, you need to enable preview features.
+
+Sahibi IMs Ainsley to ask what this is about where she learns that this feature
+isn't a supported part of .NET, therefore she should make sure to not integrate
+the library into the production instance of Fabrikam.
+
+## Requirements
+
+### Goals
+
+* Customers can discover preview features
+* Customers are made aware if a particular feature is in a preview state so that
+  they always know which part of the product is fully supported and which parts
+  aren't.
+* Customer have an easy way to audit if their applications or libraries are
+  using preview features.
+* Customers can't accidentally depend on preview features, either directly or
```

Comment:
Does this cover also depencies taken via reflection? It may be worth having a paragraph dedicated to reflection.

Consider:
```
void f(dynamic x) => x.ExperimentalMethod();
```

Reply:
I have added an open issue in the design section.